modules/siddhi-extensions/input-mappers/text-input-mapper/src/main/java/org/wso2/siddhi/extension/input/mapper/text/TextSourceMapper.java

```diff
@@ -135,13 +134,13 @@ public void init(StreamDefinition streamDefinition, OptionHolder optionHolder, L
      * {@link OutputCallback}.
      *
      * @param eventObject  the TEXT string
-     * @param inputHandler input handler
      */
     @Override
-    protected void mapAndProcess(Object eventObject, InputHandler inputHandler) throws InterruptedException {
+    protected void mapAndProcess(Object eventObject) throws InterruptedException {
         if (eventObject != null) {
             synchronized (this) {
-                inputHandler.send(convertToEvent(eventObject));
+                Event event = convertToEvent(eventObject);
+                send(event);
```

Comment:
Don't use send(), and don't implement send in SourceMapper.java. The InputHandler is passed as a parameter for a reason, such that the person extending will get a hint that they can send events to that and it will be intuitive. Please pass the InputEventHander here so the implementing developer can do inputEventHander.send()